scipy/_lib/tests/test_public_api.py

```diff
@@ -112,7 +112,6 @@ def test_dir_testing():
     'scipy.linalg.decomp_qr',
     'scipy.linalg.decomp_schur',
     'scipy.linalg.decomp_svd',
-    'scipy.linalg.flinalg',
```

Comment:
Wouldn't we have to at least create a stub module that's still importable but raises a deprecation warning

Reply:
I would like to risk that nobody on their right mind is importing det and lu from flinalg which emits deprecation warning already. 

There is always nonzero probability of course but I tend to choose not acting on it here. Purely my subjective opinion though. I can defer the removal to later version

Reply:
Yeah, I had missed that these are emitting warnings already. The warning however does pretty clearly announce removal for 2.0:
```
>>> from scipy.linalg.flinalg import *
<stdin>:1: DeprecationWarning: The `scipy.linalg.flinalg` namespace is deprecated and will be removed in SciPy v2.0.0.
```
I'm not strictly against removing it, but it's a bit of inconsistent messaging (saying it'll be removed in 2.0, then removing earlier after all)

Reply:
Somehow I managed to read it as 1.12.0... Dang.

Reply:
@rgommers @tylerjereddy On one hand indeed this is breaking contract, on the other hand it is a very obscure module that only provides det and lu fortran api. Do you think it is safe to remove or should we leave it until SciPy 2.0 indeed?

Reply:
I'd say that if it's just a matter of putting it back because you misread 2.0 as 1.12.0 and it's little effort, then put it back. But if it's more work to keep the deprecated and private API working because of other changes in this PR, then let's be pragmatic and get rid of it - this is not worth much effort either way.

Reply:
OK added flinalg back. Thanks all.